test/helpers/test.go

```diff
@@ -171,6 +177,55 @@ func (t *TestOptions) GetGitProvider() (gits.GitProvider, error) {
 	return gitProvider, nil
 }
 
+// GetApproverGitProvider returns a git provider that uses credentials for the approver user defined in environment variables
+// We don't use standard user auth here because the user/token isn't defined during boot, so we'll create the credentials
+// for the user on the fly.
+func (t *TestOptions) GetApproverGitProvider() (gits.GitProvider, error) {
+	factory := cmd.NewFactory()
+	_, ns, err := factory.CreateKubeClient()
+	if err != nil {
+		return nil, err
+	}
+
+	authConfigService, err := factory.CreateAuthConfigService("gitAuth.yaml", ns, kube.ValueKindGit, "")
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := authConfigService.LoadConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error loading auth config: %s", err)
+	}
+
+	if config == nil {
```

Comment:
Something is not right here. I mean in _jx_. What is the contract for `authConfigService.LoadConfig()`? Is _nil_ a valid return value or should there always be an AuthConfig returned (in case there was no error)? This code seems to imply both is the case which seems odd. Or we somehow use the API in an unintended way? We might need to create an issue for that in jx.

Reply:
I have no idea. Again, I just copied and tweaked the existing `GetGitProvider`.